Doc/howto/perf_profiling.rst

```diff
@@ -147,52 +148,61 @@ Instead, if we run the same experiment with perf support activated we get:
 
 
 
-Enabling perf profiling mode
-----------------------------
+How to enable ``perf`` profiling support
+----------------------------------------
 
-There are two main ways to activate the perf profiling mode. If you want it to be
-active since the start of the Python interpreter, you can use the `-Xperf` option:
+There are three ways to enable ``perf`` profiling support;
+the environment variable :envvar:`PYTHONPERFSUPPORT` and the
+:option:`-X perf <-X>` option allow you to enable perf profiling from the start,
+whereas the :func:`sys.activate_stack_trampoline` and
+:func:`sys.deactivate_stack_trampoline` functions allow you to enable and
+disable it dynamically.
 
-    $ python -Xperf my_script.py
+The :mod:`!sys` functions take precedence over the :option:`!-X` option,
+the :option:`!-X` option takes precedence over the environment variable.
 
-You can also set the :envvar:`PYTHONPERFSUPPORT` to a nonzero value to actiavate perf
-profiling mode globally.
+Example, using the environment variable::
 
-There is also support for dynamically activating and deactivating the perf
-profiling mode by using the APIs in the :mod:`sys` module:
+   $ PYTHONPERFSUPPORT=1
+   $ python script1.py
+   $ python script2.py
+   $ python script3.py
+   $ perf report -g -i perf.data
```

Comment:
This document doesn't seem to mention how `perf.data` is created and what it contains.  It would appear that is created automatically (with the `perf.data` name) whenever `perf` mode is enabled, however it is not clear if in this example the results of the three executions are combined, if they are separated within the file, or if every separate command will create a new `perf.data` that will override the previous one.

I'm also not sure if running ``perf`` on three different files like this is a realistic use case.  If not, maybe the following is better:
```suggestion
   $ PYTHONPERFSUPPORT=1 python script1.py
   $ perf report -g -i perf.data
```

Reply:
I agree that this part can be improved. Since I do not know the answers to your questions from the top of my head, I need to actually test this on a Linux system (I've only got WSL Ubuntu instance at my hands right now, and I can't get perf up and running there).